### PR TITLE
create register page #19

### PR DESCRIPTION
**Pull request by walaayousef98**

### CODE REVIEW

File: ui/src/app/Custom Validator/ConfirmPass.validator.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { FormGroup } from '@angular/forms';
```
Comment (diaahamad) [inline ui/src/app/Custom Validator/ConfirmPass.validator.ts:1]: is this function will work with any controls or just password

File: ui/src/app/app.component.ts [hunk 2]
```diff
@@ -1,9 +1,8 @@
 import { Component } from '@angular/core';
-
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.sass']
+  styleUrls: ['./app.component.sass'],
```
Comment (diaahamad) [inline ui/src/app/app.component.ts:5]: why this semicolon ?

File: ui/src/app/auth/login/login.component.html [hunk 3]
```diff
@@ -3,41 +3,51 @@
...
-        <div fxLayout="column" fxLayoutAlign="center">
+        <div fxLayout="column" fxLayoutAlign="center" >
           <form [formGroup]="loginForm" (ngSubmit)="onSubmit()">
             <div fxLayoutAlign="center">
               <mat-form-field fxFlex="60" fxFlex.lt-md="90" appearance="outline">
                 <mat-label>Email</mat-label>
-                <input formControlName="email" type="email" matInput  [class.is-invalid]="loginForm.controls['email'].invalid && loginForm.controls['email'].touched" [class.is-valid]="loginForm.controls['email'].valid" required>
-                <mat-error *ngIf="loginForm.controls['email'].invalid">Please enter a valid email</mat-error>
+                <input formControlName="email" type="email" matInput
+                  [class.is-invalid]="loginForm.controls['email'].invalid && loginForm.controls['email'].touched"
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.html:22]: we do no not need those attributes  with angular material

File: ui/src/app/auth/login/login.component.html [hunk 4]
```diff
@@ -3,41 +3,51 @@
...
+                  Please enter a valid email (example@gmail.com)
+                </mat-error>
               </mat-form-field>
             </div>
             <div fxLayoutAlign="center">
               <mat-form-field fxFlex="60" fxFlex.lt-md="90" appearance="outline">
                 <mat-label>Password</mat-label>
-                <input formControlName="password" type="password" matInput [class.is-invalid]="loginForm.controls['password'].invalid && loginForm.controls['password'].touched" [class.is-valid]="loginForm.controls['password'].valid" required>
+                <input formControlName="password" type="password" matInput
+                  [class.is-invalid]="loginForm.controls['password'].invalid && loginForm.controls['password'].touched"
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.html:36]: same here

File: ui/src/app/Custom Validator/ConfirmPass.validator.ts (hunk 1 above)
Comment (walaayousef98) [reply ui/src/app/Custom Validator/ConfirmPass.validator.ts:1]: for any two controls you want to compare them?

**Comment by walaayousef98** [comment]
Modified

File: ui/src/app/app.component.ts [hunk 5]
```diff
@@ -1,5 +1,4 @@
 import { Component } from '@angular/core';
-
```
Comment (diaahamad) [inline ui/src/app/app.component.ts:2]: why this new line

File: ui/src/app/auth/login/login.component.ts [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
   styleUrls: ['./login.component.sass']
 })
 export class LoginComponent implements OnInit {
-
+  hidePassword = true;
   loginForm : FormGroup ;
   formSubmitted : boolean = false;
 
-  constructor(private fb:FormBuilder) { }
+  constructor(private fb:FormBuilder,private route:Router) { }
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.ts:17]: add space after semicolon

File: ui/src/app/auth/register/register.component.html [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+          <div fxLayout="row" fxLayoutAlign="center">
+            <h1 class="title">Welcome To Candi Meter</h1>
+          </div>
+        </div>
+        <div fxLayout="column" fxLayoutAlign="center">
+          <form [formGroup]="registerForm" (ngSubmit)="onSubmit()">
+            <div fxLayoutAlign="center">
+              <mat-form-field fxFlex="60" fxFlex.lt-md="90" appearance="outline">
+                <mat-label>Email</mat-label>
+                <input formControlName="email" type="email" matInput autofocus required>
```
Comment (diaahamad) [inline ui/src/app/auth/register/register.component.html:23]: we do not need to add required here

File: ui/src/app/app.component.ts (hunk 5 above)
Comment (walaayousef98) [reply ui/src/app/app.component.ts:2]: I removed this line

File: ui/src/app/auth/register/register.component.html (hunk 7 above)
Comment (walaayousef98) [reply ui/src/app/auth/register/register.component.html]: done

File: ui/src/app/auth/login/login.component.ts [hunk 8]
```diff
@@ -22,7 +25,11 @@ export class LoginComponent implements OnInit {
 
   onSubmit() {
     this.formSubmitted = true;
-      if (!this.loginForm.valid) {
+    
+    if (this.loginForm.valid) {
+    this.route.navigate(["/login"]);
```
Comment (diaahamad) [inline ui/src/app/auth/login/login.component.ts:30]: use navigatebyurl

File: ui/src/app/auth/register.animation.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { animate, keyframes, state, style, transition, trigger } from "@angular/animations";
+
+export function registerScreenAnimation(){
+    return  trigger('registerScreenAnimation', [
```
Comment (diaahamad) [inline ui/src/app/auth/register.animation.ts:4]: debug here and see if this function called multi times

File: ui/src/app/auth/register/register.component.ts [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    this.registerForm = this.fb.group({
+      email: ['', [Validators.required, Validators.pattern(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/)]],
+      password: ['', [Validators.required,Validators.pattern(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/)]],
+      confirmPassword: ['', [Validators.required]]
+
+    },
+     { validator: MustMatch('password', 'confirmPassword')});
+  }
+ 
+    get email(){
```
Comment (diaahamad) [inline ui/src/app/auth/register/register.component.ts:35]: where is the using of those functions

File: ui/src/styles.sass [hunk 11]
```diff
@@ -1 +1,5 @@
 /* You can add global styles to this file, and also import other style files */
+body
+  margin: 0%
+
+@import 'material-icons/iconfont/material-icons.scss'
```
Comment (diaahamad) [inline ui/src/styles.sass:5]: the import line must be on the first of the file

File: ui/src/app/auth/register/register.component.ts (hunk 10 above)
Comment (walaayousef98) [reply ui/src/app/auth/register/register.component.ts:35]: In order to store the data that the user writes in the database and then call it in the login and compare it with the data entered

File: ui/src/styles.sass (hunk 11 above)
Comment (walaayousef98) [reply ui/src/styles.sass:5]: ok

File: ui/src/app/auth/register.animation.ts (hunk 9 above)
Comment (walaayousef98) [reply ui/src/app/auth/register.animation.ts:4]: called once when I press the register button when I have something wrong with the form

**Comment by walaayousef98** [comment]
done

**Pull request merged**